plugins/verifier/vulnerabilityreport/vulnerability_report.go

```diff
@@ -93,34 +94,46 @@ func VerifyReference(args *skel.CmdArgs, subjectReference common.Reference, refe
 	}
 	createdTime, err := extractCreationTimestamp(input.CreatedAnnotationName, referenceDescriptor)
 	if err != nil {
+		verifierErr := re.ErrorCodeVerifyPluginFailure.WithDetail("Failed to create timestamp annotation.").WithError(err)
 		return &verifier.VerifierResult{
-			Name:      input.Name,
-			Type:      verifierType,
-			IsSuccess: false,
-			Message:   fmt.Sprintf("Validation failed: error extracting create timestamp annotation:[%v]", err.Error()),
+			Name:         input.Name,
+			Type:         verifierType,
+			VerifierName: input.Name,
+			VerifierType: verifierType,
+			IsSuccess:    false,
+			Message:      verifierErr.GetDetail(),
```

Comment:
Can construction of VerifierResult be refactored into a common method? then as we add new/remove properties, we only have to update it in the single place.

Reply:
That's good call! I'll make the refactoring.